packages/sdk/src/common/logger/logger-middleware.ts

```diff
@@ -0,0 +1,87 @@
+import { Blockchain } from "@rarible/api-client"
+import type { BlockchainWallet } from "@rarible/sdk-wallet"
+import { RemoteLogger } from "@rarible/logger/build"
+import type { LoggableValue } from "@rarible/logger/build/domain"
+import axios from "axios"
+import type { Middleware } from "../middleware/middleware"
+import type { ISdkContext } from "../../domain"
+
+type LogsLevel = "trace" | "error" | "disabled"
+
+const loggerConfig = {
+	ELK_URL: "https://logging.rarible.com/",
+}
+
+async function getWalletInfo(wallet: BlockchainWallet): Promise<Record<string, string>> {
+	const info: Record<string, any> = {
+		"wallet.blockchain": wallet.blockchain,
+	}
+
+	switch (wallet.blockchain) {
```

Comment:
Instead of switch casing it'll be more clear so something like

```
const context = {
 address: await getAddress(wallet),
 chainId: await getChainId(wallet)
}
```

And you'll be able to cover your context with types in this case (not just Record<string, any>

Reply:
Not sure if decomposition of 1 switch/case will be good choice

1. Switch/case just moves to utility funcs (getAddress, getChainId), and here switch logic code duplicates across these functions
2. `const context = {
 address: await getAddress(wallet),
 chainId: await getChainId(wallet)
}` makes serial calls instead of parallel

I guess right way is provide universal address/network Id getters in Blockchain Wallet, since this pretty common use cases